### ISSUE TITLE
stop "jumping" to top of page when savoing content in a "card"

### ISSUE THREAD

**Issue by Kodzhabashev** [humdek-unibe-ch/sh-selfhelp#197]
In GitLab by @juliahegy on Oct 3, 2019, 15:39

When our test users saved their input in a "card" the page always "jumped" to the very top. They found this to be frustrating and not very userfriendly. 
Is there a possibility to fix this, i.e., to stop the "jumping"?

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Oct 3, 2019, 16:07

What do you mean by a "card"?
Are you referring to a `formUserInput`?
If so this should have been fixed a long time ago (version 0.9.12) in issue #140.

Can you please provide me with a link to an example where this behaviour occurs.

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Oct 4, 2019, 09:51

Comment from @juliahegy 
>  I am referring to the style “card”.  An example: In the SelFIT tool, Modul 2, we used a card for the exercise “Irrationale Annahmen erkennen und hinterfragen” (Link: https://selfhelp.psy.unibe.ch/SelFIT/Modul2)
![image](http://phhum-a209-cp.unibe.ch:10012/SLP/SLP-sleep_coach/uploads/90e9a622e913482e31a9b89410f81429/image.png)

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Oct 4, 2019, 09:55

The problem is that you placed the form in a collapsible card.
This means that when the browser wants to jump to the anchor of the form it cannot because the form is not rendered.

Right now you can solve this by not making the card collapsible.
For a future release we will be able to add a new field to the style `formUserInput` which would allow you to specify a custom anchor (e.g. the section name of the card wrapping the form) to which the browser should jump on submit.

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Oct 4, 2019, 10:33

Allow me to make a side note here:

I believe that features such as jumping back to a form and the new button that allows to jump to the top of a page (#191) should only be necessary in some rare cases. In general user navigation should not rely on this. The example link you provided, for example, is a very long page that requires a lot of scrolling. I am wondering if the navigation might me more intuitive to a user if this page would be split up into navigation pages. As an example you can refer to the [styles documentation page](https://selfhelp.psy.unibe.ch/demo/style/445).
Note that when you scroll down, the navigation menu follows the scrolling.
I did this with the following custom css class (which I added to the field `css_nav` of the navigation page)
```css
.nav-menu
{
    position: -webkit-sticky;
    position: sticky;
    top: 1rem;
    z-index: 900;
}
```

**Comment by Kodzhabashev** [comment]
In GitLab by @juliahegy on Oct 4, 2019, 11:28

Makes sense. Can you tell us when this future release will take place (approximately)? 
Thank you :)

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Oct 4, 2019, 12:24

All TPF members are currently working on different projects. This feature will be implemented (along with other requested features) as soon as the TPF has received a new application and the TPF board has scheduled the project accordingly. Also see my comments in #196.

After the 22nd of October I will know more.

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Apr 29, 2020, 15:04

mentioned in commit a5a762af3b762b0e3b71094c5cb5dc81fc63a4cd

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Apr 29, 2020, 16:35

closed via commit a5a762af3b762b0e3b71094c5cb5dc81fc63a4cd

**Comment by Kodzhabashev** [comment]
In GitLab by @moiri on Apr 29, 2020, 17:10

mentioned in commit 7400758c411a95883fd6569be19ae9ed544aba56
